pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1270,6 +1270,9 @@ func updateNonPinnedCPUs(ctx *domainContext, config *types.DomainConfig, status
 	return nil
 }
 
+// assignCPUs assigns CPUs to the VM based on the configuration
+// By the assignment, we mean that the CPUs are assigned in the CPUAllocator context to the given VM
+// and the cpumask is updated in the *status*
 func assignCPUs(ctx *domainContext, config *types.DomainConfig, status *types.DomainStatus) error {
 	if config.VmConfig.CPUsPinned { // Pin the CPU
```

Comment:
Similar to the free, can this be called more than once and hence we should make it check whether it already allocated some cpusToAssign for this app instance?

Reply:
It was the original error: we called it more than once for the same domain. No, it should not be called more than once. If the domain is not going to use the CPUs immediately (it is not ready to finish the activation), I expect it to free the CPUs. It's the idea of this PR.
And I don't want to make the attempt to call `assignCPUs` several times silent: it can lead to a resource leak.